README.md

```diff
@@ -320,98 +794,368 @@ And configure your remote endpoint:
+    middlewares.parse.push(cookieParser());
+  }
+}
+
+module.exports = MyExpressMixin;
+```
+
+### Configuring rendering through runtime mixins
+
+Hops provides multiple hooks to customize rendering. We'll only show an example for the `enhanceElement` hook here, but encourage you to browse through the source code of Hops to find other runtime (or browser/server) mixins to see examples of these hooks in action.
+
+The `enhanceElement(reactElement)` hook will be called in a [functioncal compositaion](<https://en.wikipedia.org/wiki/Function_composition_(computer_science)>) manner and therefore accepts a single argument of the type of a React element and expects you to return a React element which wraps the input.
+
+The following example demonstrates how you could implement a custom [mobx](https://mobx.js.org/) preset:
+
+**`mixin.runtime.js`**
+
+```javascript
+const { Mixin } = require('hops');
+const { Provider } = require('mobx-react');
+const React = require('react');
+
+class MyMobxMixin extends Mixin {
+  constructor(config, element, { mobx: options = {} } = {}) {
+    super(config);
+    this.stores = options.stores;
+  }
+
+  enhanceElement(reactElement) {
+    return <Provider {...this.stores}>{reactElement}</Provider>;
+  }
+}
+
+module.exports = MyMobxMixin;
 ```
 
-Afterwards, run `yarn install` to install the required dependencies.
+And then, once [activated](#activating-presets), you can use it like this in your application:
 
-Using `yarn start` will execute the `start` script in the [hops-template-react](https://github.com/xing/hops/tree/master/packages/template-react) so that you can verify your changes in the browser.
+**`src/app.js`**
 
-Hops is written using modern ECMAScript features (ES 2017) and uses [CommonJS](https://nodejs.org/docs/latest/api/modules.html) as its module system. The code is formatted using [prettier](https://prettier.io) and checked via [ESlint](https://eslint.org/) and [Jest](https://jestjs.io/). You can check your code by executing: `yarn lint`, `yarn test` and format it using `yarn fmt`.
+```javascript
+import { render } from 'hops';
+import { action, observable } from 'mobx';
+import { inject, observer } from 'mobx-react';
+import React from 'react';
+
+const theme = observable(
+  {
+    color: 'red',
+    setColor(color) {
+      this.color = color;
+    },
+  },
+  { setColor: action }
+);
+
+const MyApp = inject('theme')(
+  observer(props => (
+    <h1
+      style={{ color: props.theme.color }}
+      onClick={() => props.theme.setColor('green')}
+    >
+      hello world
+    </h1>
+  ))
+);
+
+export default render(<MyApp />, { mobx: { stores: { theme } } });
+```
+
+### Presets
+
+Presets are being used to provide reusable / shareable [settings](#settings). Optionally they can also provide and configure their own mixins.
+
+A preset can either be written as JavaScript, JSON or yaml file (`preset.{js,json,yml}`) or can be specified in the `"preset"` field in a `package.json`.
```

Comment:
I am afraid that to be found without being specified in the `preset` field in package.json, the file needs to be called `preset.js`. Alternatively, all valid config file names are also valid here.

Reply:
I removed the part about json, yaml, etc.